src/main/java/school/faang/bjs245128/Main.java

```diff
@@ -0,0 +1,68 @@
+package school.faang.bjs245128;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+public class Main {
+
+    public static HashSet<Product> products = new HashSet<>();
```

Comment:
Это тоже не ошибка, но тем не менее, напишу как рекомендацию. В java принято оперировать абстракциями, а не реализациями (собственно, потому что это ООП язык). Это нам позволяет быть более гибким и не зависеть от реализации. Поэтому, когда создают переменные или когда создают методы с входными параметрами, по возможности всегда используют абстракции. То есть в данном случае я бы рекомендовал в методах и в объявлении переменной использовать просто Set:
`
public static Set<Product> products = new HashSet<>();
`

Reply:
Dear Alexey I know this. But according to task it says create exactly as here. Of course I prefer abstraction and loosely coupling rather than tightly coupling. But again thanks